Bottom: 2a552ac0cc5f1859f6e05ec0bd0e1673a2c2a8d7
Top:    3a14d9ab482bda4efe85623436a9ddbb2a9aaad0
Author: Ian Kent <raven@themaw.net>
Date:   2013-05-02 15:53:54 +0800

Refresh of autofs-5.0.7-use-MS_UNBINDABLE-for-autofs-mounts.patch

---

diff --git a/daemon/direct.c b/daemon/direct.c
index b1fd74a..38c8543 100644
--- a/daemon/direct.c
+++ b/daemon/direct.c
@@ -420,7 +420,7 @@ int do_mount_autofs_direct(struct autofs_point *ap,
 
 	map_name = me->mc->map->argv[0];
 
-	ret = mount(map_name, me->key, "autofs", MS_UNBINDABLE|MS_MGC_VAL, mp->options);
+	ret = mount(map_name, me->key, "autofs", mountflags, mp->options);
 	if (ret) {
 		crit(ap->logopt, "failed to mount autofs path %s", me->key);
 		goto out_err;
@@ -768,7 +768,7 @@ int mount_autofs_offset(struct autofs_point *ap, struct mapent *me, const char *
 	if (!type || strcmp(ap->entry->maps->type, "hosts"))
 		map_name = me->mc->map->argv[0];
 
-	ret = mount(map_name, mountpoint, "autofs", MS_UNBINDABLE|MS_MGC_VAL, mp->options);
+	ret = mount(map_name, mountpoint, "autofs", mountflags, mp->options);
 	if (ret) {
 		crit(ap->logopt,
 		     "failed to mount offset trigger %s at %s",
diff --git a/daemon/indirect.c b/daemon/indirect.c
index 2b0ea6e..45758ef 100644
--- a/daemon/indirect.c
+++ b/daemon/indirect.c
@@ -146,7 +146,7 @@ static int do_mount_autofs_indirect(struct autofs_point *ap, const char *root)
 	if (!type || strcmp(ap->entry->maps->type, "hosts"))
 		map_name = ap->entry->maps->argv[0];
 
-	ret = mount(map_name, root, "autofs", MS_UNBINDABLE|MS_MGC_VAL, options);
+	ret = mount(map_name, root, "autofs", mountflags, options);
 	if (ret) {
 		crit(ap->logopt,
 		     "failed to mount autofs path %s at %s", ap->path, root);
